pkg/api/v20240812preview/openshiftcluster.go

```diff
@@ -0,0 +1,303 @@
+package v20240812preview
+
+import "time"
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+// OpenShiftClusterList represents a list of OpenShift clusters.
+type OpenShiftClusterList struct {
+	// The list of OpenShift clusters.
+	OpenShiftClusters []*OpenShiftCluster `json:"value"`
+
+	// The link used to get the next page of operations.
+	NextLink string `json:"nextLink,omitempty"`
+}
+
+// OpenShiftCluster represents an Azure Red Hat OpenShift cluster.
+type OpenShiftCluster struct {
+	// The resource ID.
+	ID string `json:"id,omitempty" mutable:"case"`
+
+	// The resource name.
+	Name string `json:"name,omitempty" mutable:"case"`
+
+	// The resource type.
+	Type string `json:"type,omitempty" mutable:"case"`
+
+	// The resource location.
+	Location string `json:"location,omitempty"`
+
+	// SystemData - The system metadata relating to this resource
+	SystemData *SystemData `json:"systemData,omitempty" swagger:"readOnly"`
+
+	// The resource tags.
+	Tags Tags `json:"tags,omitempty" mutable:"true"`
+
+	// The cluster properties.
+	Properties OpenShiftClusterProperties `json:"properties,omitempty"`
+}
+
+// Tags represents an OpenShift cluster's tags.
+type Tags map[string]string
+
+// OpenShiftClusterProperties represents an OpenShift cluster's properties.
+type OpenShiftClusterProperties struct {
+	// The cluster provisioning state.
+	ProvisioningState ProvisioningState `json:"provisioningState,omitempty"`
+
+	// The cluster profile.
+	ClusterProfile ClusterProfile `json:"clusterProfile,omitempty"`
+
+	// The console profile.
+	ConsoleProfile ConsoleProfile `json:"consoleProfile,omitempty"`
+
+	// The cluster service principal profile.
+	ServicePrincipalProfile ServicePrincipalProfile `json:"servicePrincipalProfile,omitempty"`
+
+	// The cluster network profile.
+	NetworkProfile NetworkProfile `json:"networkProfile,omitempty"`
+
+	// The cluster master profile.
+	MasterProfile MasterProfile `json:"masterProfile,omitempty"`
+
+	// The cluster worker profiles.
+	WorkerProfiles []WorkerProfile `json:"workerProfiles,omitempty"`
+
+	// The cluster worker profiles status.
+	WorkerProfilesStatus []WorkerProfile `json:"workerProfilesStatus,omitempty" swagger:"readOnly"`
+
+	// The cluster API server profile.
+	APIServerProfile APIServerProfile `json:"apiserverProfile,omitempty"`
+
+	// The cluster ingress profiles.
+	IngressProfiles []IngressProfile `json:"ingressProfiles,omitempty"`
+}
+
+// ProvisioningState represents a provisioning state.
+type ProvisioningState string
+
+// ProvisioningState constants.
+// TODO: ProvisioningStateCancelled is included to pass upstream CI. It is currently unused in ARO.
+const (
+	ProvisioningStateCreating      ProvisioningState = "Creating"
+	ProvisioningStateUpdating      ProvisioningState = "Updating"
+	ProvisioningStateCancelled     ProvisioningState = "Canceled"
```

Comment:
Not sure what the preference is here, but in v20231122 I updated the comment and variable names to 'Canceled' as well.

Reply:
We should ensure the same approach is taken on both APIs - in the small chance we have to cut a new stable API release off of 20231122 while 20230812preview is still active, we'd end up having to merge diffs between both APIs, just as we did on 20231122 itself. Keeping things as close together as possible will help make that process less painful.

Reply:
Oops, this was an oversight and wasn't an intentional choice. I'll fix it. 🙂

Reply:
Hey - unsure what happened but this `const` name got reverted.

Reply:
Should be fixed now!